### PR TITLE
add format parser

### PR DESCRIPTION
**Pull request by Jiramew**
Add parser to extract value from format. 
Default parser will be:
```js
string.replace(/[^\w\.-]*/g, '') 
```

fix https://github.com/ant-design/ant-design/pull/5178

Thanks so much
Jiramew

### CODE REVIEW

File: src/mixin.js [hunk 1]
```diff
@@ -56,7 +56,13 @@ export default {
   },
 
   onChange(e) {
-    const input = this.getValueFromEvent(e).trim().replace(/^[^\w\.-]*|[^\w\.-]*$/g, '');
+    let input = this.getValueFromEvent(e).trim();
+    if (this.props.parser) {
+      input = this.props.parser(input);
+    } else {
+      input = input.replace(/[^\w\.-]*/g, '');
```
Comment (paranoidjk) [inline src/mixin.js]: maybe set `defaultProps.parser` is better

Comment (Jiramew) [inline src/mixin.js]: Thanks so much~ (^^)

File: README.md [hunk 2]
```diff
@@ -205,6 +205,12 @@ react-native run-ios
           <td></td>
           <td>Specifies the format of the value presented</td>
         </tr>
+        <tr>
+          <td>parser</td>
+          <td>Function</td>
+          <td></td>
```
Comment (paranoidjk) [inline README.md]: parser default value should be show in document

**Comment by paranoidjk** [comment]
always, it's better to use `rebase -i`  to merge your multiple sommit before `git push ` 😄

**Comment by afc163** [comment]
Maybe renamed `parse` to `unformatter` would be more clear?

**Comment by afc163** [comment]
Since `formatter` and `parser` basicly must be defined at same time. How about this one?

```js
function formatter(value, direction) {
  if (direction === 'format') {
    return `$ ${this.numberWithCommas(value)} boeing737`; 
  } else if (direction === 'parse') {  // reverse formatting
    return value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ','); 
  }
}
```

**Comment by paranoidjk** [comment]
maybe like this

```js
props.transform = {
 formatter: value => value, // default maybe
 parse: value => value
}
```

**Pull request merged**